### DISCUSSION TITLE
Generate multiple relation field from many to may in view collection

### DISCUSSION THREAD

**Discussion by sergio9929** [pocketbase/pocketbase#3194] [Q&A] [answered]
I was wondering if there is a way to generate a multiple relation field using a many to many table in a collection of type view.

Given this table (taken from the docs https://pocketbase.io/docs/working-with-relations/):

![image](https://github.com/pocketbase/pocketbase/assets/29752861/366ebb5a-7ea6-408a-86bc-51f24f42dcf1)

What I've tried:

```sql
SELECT
  posts.id as id,
  posts.title as title,
  count(comments.id) as commentCount,
  comments.id as comments
FROM posts
LEFT JOIN comments on posts.id =comments.post
GROUP BY posts.id
```

Expected result:

title (text) | commentCount (number) | comments (relation)
----------- | ---------------------------- | ---------------------
Example 1 | 3                                       | comment1, comment2, comment3
Example 2 | 2                                       | comment4, comment5

My real query:

```sql
SELECT
  incidences.id as id,
  incidences.code as code,
  count(actionsIncidences.id) as actionCount,
  actionsIncidences.id as actions
FROM incidences
LEFT JOIN actionsIncidences on incidences.id = actionsIncidences.incidenceId
GROUP BY incidences.id
```

**Comment by ganigeorgiev** [comment, marked as answer]
Unfortunately this is not support at the moment.

_There are plans in the future to allow users to provide custom fields mapping/definition for the view query in addition to the auto detected fields ([#roadmap32302192](https://github.com/orgs/pocketbase/projects/2/views/1?pane=issue&itemId=32302192)), but even when the extra mapping is implemented the above 2 queries will not work because for the multiple `relation` field value you'll need the result column to be a serialized json array with the grouped relation ids (this array could be generated for example with [json_group_array](https://www.sqlite.org/json1.html#jgrouparray))._

**Reply by sergio9929** [reply]
Thank you very much for the quick response.
